multiModule1/subMultiModule1/src/main/java/com/example/api/APIQueryHandler.java

```diff
@@ -0,0 +1,91 @@
+package com.example.api;
+
+import com.example.data.ConfigData;
+
+import java.util.List;
+import java.util.concurrent.locks.Condition;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+
+public class APIQueryHandler {
+    private ConfigData[] configs;
+    private List<String> outputs;
+    private static Lock LOCK = new ReentrantLock(true);
+
+    private volatile int requestCounter = 0;
+
+    public Lock getLock() {
+        return LOCK;
+    }
+
+    public ConfigData[] getConfigs() {
+        return configs;
+    }
+
+    public synchronized void setConfigs(ConfigData[] configs) {
+        this.configs = configs;
+    }
+
+    public List<String> getOutputs() {
+        return outputs;
+    }
+
+    public APIQueryHandler(ConfigData[] configs) {
+        this.configs = configs;
+    }
+
+    /**
+     * Shortcut for calling wait
+     */
+    private void waitForLock(Condition c) {
+        try {
+            c.wait();
+        } catch (Throwable e) {}
+    }
+
+    /**
+     * Performs network queries in parallel and retrieves the info.
+     * @return The list of results.
+     * @throws InterruptedException
+     */
+    public void getDataInParallel() throws InterruptedException {
+
+        Thread[] ts = new Thread[configs.length];
+
+        // Locks make use of condition variables for synchronization.
+        Condition prevDone = LOCK.newCondition();
+        for (int i = 0; i < configs.length; ++i) {
+            int finalI = i;
+            ts[i] = new Thread(
+                    () -> {
```

Comment:
Is this a valid pattern in Java, or is it meant to raise issues?

Reply:
it was an ide suggestion